### PR TITLE
amber: init at 0.1.1

### PR DESCRIPTION
**Pull request by psibi**
###### Motivation for this change

This tools allows for easily managing secrets in a repo. Tested it locally on a NixOS machine:

```
❯ amber
amber 0.1.1
Utility to store encrypted secrets in version trackable plain text files

USAGE:
    amber [FLAGS] [OPTIONS] <SUBCOMMAND>

FLAGS:
    -h, --help        Prints help information
        --unmasked    Disable masking of secret values during exec
    -v, --verbose     Turn on verbose output
    -V, --version     Prints version information

OPTIONS:
        --amber-yaml <amber-yaml>    amber.yaml file location [env: AMBER_YAML=] [default:
                                     amber.yaml]

SUBCOMMANDS:
    encrypt     Add or update a secret
    exec        Run a command with all of the secrets set as environment variables
    generate    Generate a new strong secret value, and add it to the repository
    help        Prints this message or the help of the given subcommand(s)
    init        Initialize a new directory
    print       Print all of the secrets
    remove      Remove a secret
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/amber/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    sha256 = "1l5c7vdi885z56nqqbm4sw9hvqk3rfzm0mgcwk5cbwjlrz7yjq4m";
+  };
+
+  cargoSha256 = "sha256:0dmhlyrw6yd7p80v7anz5nrd28bcrhq27vzy605dinddvncjn13q";
+
+  meta = with lib; {
+    description = "Manage secret values in-repo via public key cryptography";
+    homepage = "https://github.com/fpco/amber";
+    license = licenses.mit;
+    maintainers = [ maintainers.psibi ];
```
Comment (figsoda) [inline pkgs/tools/security/amber/default.nix:21]:
```suggestion
    maintainers = with maintainers; [ psibi ];
    mainProgram = "amber";
```

**Review by figsoda — CHANGES_REQUESTED** [review]
package is not added to all-packages.nix, also commit message should be `amber-secret: init at 0.1.1`

File: pkgs/tools/security/amber/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "fpco";
+    repo = "amber";
+    rev = "v${version}";
+    sha256 = "1l5c7vdi885z56nqqbm4sw9hvqk3rfzm0mgcwk5cbwjlrz7yjq4m";
+  };
+
+  cargoSha256 = "sha256:0dmhlyrw6yd7p80v7anz5nrd28bcrhq27vzy605dinddvncjn13q";
```
Comment (SuperSandro2000) [inline pkgs/tools/security/amber/default.nix:15]:
```suggestion
  cargoSha256 = "0dmhlyrw6yd7p80v7anz5nrd28bcrhq27vzy605dinddvncjn13q";
```

File: pkgs/tools/security/amber/default.nix (hunk 1 above)
Comment (psibi) [reply pkgs/tools/security/amber/default.nix]: @figsoda Thanks for the catch! Updated and resolved them.

File: pkgs/tools/security/amber/default.nix (hunk 2 above)
Comment (psibi) [reply pkgs/tools/security/amber/default.nix]: Updated and fixed them. Thanks!

**Review by figsoda — APPROVED** [review]
maybe the file name should be amber-secret instead of amber? otherwise lgtm

**Comment by psibi** [comment]
Thanks @figsoda for the review!

> maybe the file name should be amber-secret instead of amber?

I can change that if you feel strongly. But the reason I have kept it as amber is because:

- amber is the name of the project
- I'm using `amber-secret` because there is already another package named `amber` in the nixpkgs repo.

**Comment by figsoda** [comment]
fair enough

**Comment by figsoda** [comment]
darwin build failed, i think you need `CoreFoundation` and `Security` in `buildInputs`, inherited from `darwin.apple_sdk.frameworks` inside `all-packages.nix`

**Comment by figsoda** [comment]
@ofborg build amber-secret

**Pull request merged**